.github/workflows/build-and-test-app.yml

```diff
@@ -0,0 +1,77 @@
+name: build-and-test
+run-name: build app from the ${{ github.head_ref }} branch
+on:
+  pull_request:
+  workflow_dispatch:
```

Comment:
Why do we need to trigger it manually? If I understand correctly we can go to the action tab and run this workflow, but I do not understand why we need this. Maybe, I miss something
https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow

Reply:
Why not? If you want to run the workflow to check it or for some reason, you will just run it manually.